packages/compass-editor/src/types.d.ts

```diff
@@ -0,0 +1,58 @@
+import type AceBuilds from 'ace-builds';
+import type { Ace } from 'ace-builds';
+
+type GlobalAce = {
+  EditSession: typeof AceBuilds['EditSession'];
+  Range: typeof AceBuilds['Range'];
+  UndoManager: typeof AceBuilds['UndoManager'];
+  VirtualRenderer: typeof AceBuilds['VirtualRenderer'];
+  config: typeof AceBuilds['config'];
+  createEditSession: typeof AceBuilds['createEditSession'];
+  edit: typeof AceBuilds['edit'];
+  require: typeof AceBuilds['require'];
+  version: typeof AceBuilds['version'];
+  acequire: typeof AceBuilds['require'];
+  define: (
+    module: string,
+    depdendencies: string[],
+    moduleFn: (
+      acequire: typeof AceBuilds['require'],
+      exports: any,
+      module: any
+    ) => void
+  ) => void;
+};
+
+declare global {
+  declare const ace: GlobalAce;
+}
+
+export interface HighlightRules {
```

Comment:
These are missing from `ace-build` or from `@types/ace`. I added the bare minimum we need for our modes